File: slack-reader/src/test/java/dev/number6/slackreader/dagger/RecordingSlackPortModule.java
package dev.number6.slackreader.dagger;

import dev.number6.slack.port.HttpPort;
import dev.number6.slackreader.port.SlackPort;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class RecordingSlackPortModule {

    @Provides
    @Singleton
    SlackPort providesSlackPort(HttpPort client) {
        return new RecordingSlackReaderAdaptor(client);
    }
}

File: slack-reader/src/test/kotlin/dev/number6/slackreader/dagger/RecordingSlackPortModule.kt
package dev.number6.slackreader.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slack.adaptor.SlackClientAdaptor
import dev.number6.slackreader.port.SlackReaderPort
import javax.inject.Singleton

@Module
class RecordingSlackPortModule {
    @Provides
    @Singleton
    fun providesSlackPort(client: SlackClientAdaptor): SlackReaderPort {
        return RecordingSlackReaderAdaptor(client)
    }
}